addons/sale/models/sale_order_line.py

```diff
@@ -290,6 +291,14 @@ def _compute_product_template_id(self):
     def _search_product_template_id(self, operator, value):
         return [('product_id.product_tmpl_id', operator, value)]
 
+    @api.depends('product_id.active', 'order_id.state')
```

Comment:
```suggestion
    @api.depends('product_id', 'order_id.state')
```
Since the field `has_archived_product` is not stored, there is no need to add a dependency on `active`, since the customer won't be able to archive the product directly on the SO view.
Furthermore, avoiding 'useless' dependency will reduce the work done by the ORM when the dependency is changed.  Maybe it's not the case anymore, but at least in the past, if you archived a given product, the ORM would have found all the SOL using that product, to specify that the field had to be recomputed.  Even though the field would not be recomputed in the end (because the orm would know that it didn't have to be recomputed because not stored) searching for all SOL already would have cost some 'useless' processing time.